ds3-interfaces/src/main/java/com/spectralogic/ds3client/exceptions/AggregateException.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * ****************************************************************************
+ *    Copyright 2014-2016 Spectra Logic Corporation. All Rights Reserved.
+ *    Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *    this file except in compliance with the License. A copy of the License is located at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    or in the "license" file accompanying this file.
+ *    This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *    specific language governing permissions and limitations under the License.
+ *  ****************************************************************************
+ */
+
+package com.spectralogic.ds3client.exceptions;
+
+import org.apache.commons.io.output.StringBuilderWriter;
+
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AggregateException extends RuntimeException {
+
+    private final List<Throwable> exceptions;
+
+    public AggregateException(final Iterable<Throwable> exceptions) {
+        super("One or more exceptions were aggregated together");
+        this.exceptions = exceptionList(exceptions);
+    }
+
+    private static List<Throwable> exceptionList(final Iterable<Throwable> exceptions) {
```

Comment:
I'm not sure I understand the need to make a mutable copy of the iterable contents, since the getter returns an iterable.  Though it's most likely not a big deal in terms of the amount of space used to make a copy, I'm wondering about the immutable -> mutable ness.